sisl/grid.py

```diff
@@ -180,6 +180,68 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
         # Apply the scipy.ndimage.zoom function and return a new grid
         return self.apply(ndimage_zoom, zoom_factors, mode=mode, order=order, **kwargs)
 
+    def isosurface(self, level=None, step_size=1, **kwargs):
+        """Calculates the isosurface for a given value.
+        
+        It uses skimage.measure.marching_cubes, so you need to have scikit-image installed.
+
+        Parameters
+        ----------
+        level: float
+            contour value to search for isosurfaces in the grid. 
+            If not given or None, the average of the min and max of the grid is used.
+        step_size: int
+            step size in voxels. Larger steps yield faster but coarser results. 
+            The result will always be topologically correct though.
+        **kwargs:
+            optional arguments passed directly to `skimage.measure.marching_cubes`
+            for the calculation of isosurfaces.
+        
+        Returns
+        ----------
+        numpy array of shape (V, 3)
+            Verts. Spatial coordinates for V unique mesh vertices.
+
+        numpy array of shape (n_faces, 3)
+            Faces. Define triangular faces via referencing vertex indices from verts. 
+            This algorithm specifically outputs triangles, so each face has exactly three indices.
+
+        numpy array of shape (V, 3)
+            Normals. The normal direction at each vertex, as calculated from the data.
+
+        numpy array of shape (V, 3)
+            Values. Gives a measure for the maximum value of the data in the local region near each vertex. 
+            This can be used by visualization tools to apply a colormap to the mesh.
+
+        See Also
+        --------
+        skimage.measure.marching_cubes : method used for interpolation
+        
+        """
+        try:
+            import skimage.measure
+        except:
+            raise ModuleNotFoundError("You need to have scikit-image installed in order to calculate isosurfaces.")
```

Comment:
Change to:

`f"{self.__class__.__name__}.isosurface requires scikit-image to be installed"`